File: tests/perf/test-call-basic-1.pl
sub test {
    my $i;

    sub f { return; }

    $i = 0;
    for ($i = 0; $i < 1e7; $i++) {
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
        f();
    }
}

test();

File: tests/perf/test-call-basic-1.lua
function test()
    local function f()
        return
    end

    for i=1,1e6 do
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
        f(); f(); f(); f(); f(); f(); f(); f(); f(); f()
    end
end

test()
